src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -144,14 +144,28 @@ public static string GetEffectiveConfigAsString(string gitDirectory, IFileSystem
             return stringBuilder.ToString();
         }
 
-        static string GetConfigFilePath(string workingDirectory)
+        static string GetConfigFilePath(string workingDirectory, IFileSystem fileSystem)
         {
-            return Path.Combine(workingDirectory, "GitVersionConfig.yaml");
+            var ymlPath = Path.Combine(workingDirectory, "GitVersion.yml");
+            if (fileSystem.Exists(ymlPath))
+            {
+                return ymlPath;
+            }
+
+            var yamlPath = Path.Combine(workingDirectory, "GitVersionConfig.yaml");
+            if (fileSystem.Exists(yamlPath))
+            {
+                Logger.WriteWarning("'GitVersionConfig.yaml' is deprecated, use 'GitVersion.yml' instead.");
+
+                return yamlPath;
+            }
+
+            return ymlPath;
```

Comment:
The two last returns here are redundant. IMHO it would be clearer to return `string.empty` or `null` if none of both files exists, and add checks outside, instead of returning the path to the deprecated file which not even exists. Or otherwise I would suggest to at least return the path to the current file and not the deprecated one.